### PR TITLE
some trivial cleanup

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: include/libtorrent/alert.hpp [hunk 1]
```diff
@@ -250,11 +247,6 @@ namespace libtorrent {
 
 #ifndef TORRENT_NO_DEPRECATE
 
-#ifdef __GNUC__
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```
Comment (arvidn) [inline include/libtorrent/alert.hpp]: this was to cover the use of std::auto_ptr. are they all gone?

Comment (aldenml) [inline include/libtorrent/alert.hpp]: at least in these files, yes

File: include/libtorrent/alert_manager.hpp [hunk 2]
```diff
@@ -53,12 +50,6 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #include "libtorrent/aux_/disable_warnings_pop.hpp"
 
-#ifdef __GNUC__
-// this is to suppress the warnings for using std::auto_ptr
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```
Comment (arvidn) [inline include/libtorrent/alert_manager.hpp]: I take it all mentions of std::auto_ptr<> are gone from this file too then?

Comment (aldenml) [inline include/libtorrent/alert_manager.hpp]: correct

**Comment by aldenml** [comment]
I have two questions:

1 - do you want to move to `std::function`? (with the missing `clear()` method)
2 - do you want to move to `std::shared_ptr`?

**Pull request merged**